aptos-move/aptos-vm-types/src/change_set.rs

```diff
@@ -706,6 +739,32 @@ impl VMChangeSet {
         Ok(())
     }
 
+    fn squash_additional_group_reads_needing_exchange<K>(
```

Comment:
We can have `squash_additional_reads_needing_exchange` and `squash_additional_group_reads_needing_exchange` as a single function, if we hide `&mut BTreeMap<StateKey, (WriteOp, Arc<MoveTypeLayout>)>` and `&mut BTreeMap<StateKey, (WriteOp, u64)>` under `&mut BTreeMap<StateKey, E>` for example.